utils/HtmlStructures.ts

```diff
@@ -0,0 +1,51 @@
+import { CurrentButton, User } from "../src/UserApp"
+
+
+
+
+// The structure of divs in the app
+export const DivsStructure = ( clickedBtn: CurrentButton, users: Array<User>, removedUser: Array<User>, ifSbmtClicked: boolean, ifExecutedRight: boolean ) => {
+    let output: string
+    let content: Array<string>
+    const formalOutput = ( content: string ) => QueryResult( QueryResultContent( content ) )
+    const standardStrctr: string = ( SearchBtn( clickedBtn ) + " " + AddBtn( clickedBtn ) + " " + RemoveBtn( clickedBtn ) + " " + ShowAllBtn( clickedBtn ) + Space() )
+    const inputField = InputSubmitElems( "Put your input here, please", clickedBtn )
+    const table = ( clickedBtn, users, ifSbmtClicked ) => HiddenDiv( clickedBtn, users, ifSbmtClicked, Table( TheadName( "Name" ), TheadName( "Id" ), TheadName( "Date of birth" ), Tbody( users ) ) )
+    if ( clickedBtn === CurrentButton.Search )
+        content = [standardStrctr, inputField]
+    if ( clickedBtn === CurrentButton.Search && ifSbmtClicked === true && users.length > 0 )
+        content = [standardStrctr, inputField, table( clickedBtn, users, ifSbmtClicked )]
+    if ( clickedBtn === CurrentButton.Search && ifSbmtClicked === true && users.length === 0 )
+        content = [standardStrctr, inputField, formalOutput( "No users found!" )]
+
+    if ( clickedBtn === CurrentButton.Add )
+        content = [standardStrctr, inputField, formalOutput( "" )]
+    if ( clickedBtn === CurrentButton.Add && ifSbmtClicked === true && ifExecutedRight === true ) {
+        output = "Added user: ID: " + users.indexOf( users[users.length - 1] ) + "; Full Name: " + users[users.length - 1].name + " " + users[users.length - 1].surname + "; Born in: " + users[users.length - 1].dateOfBirth + ";"
+        content = [standardStrctr, inputField, formalOutput( output )]
+    } else if ( ifExecutedRight === false )
+        content = [standardStrctr, inputField, formalOutput( "Please, write name, surname, date of birth of user" )]
+
+    if ( clickedBtn === CurrentButton.Remove )
+        content = [standardStrctr, inputField, formalOutput( "" )]
+    if ( clickedBtn === CurrentButton.Remove && ifSbmtClicked === true && ifExecutedRight === true ) {
+        output = "Removed user: Full Name: " + removedUser[0].name + " " + removedUser[0].surname + "; Born in: " + removedUser[0].dateOfBirth + ";"
+        content = [standardStrctr, inputField, formalOutput( output )]
+    } else if ( ifExecutedRight === false )
+        content = [standardStrctr, inputField, formalOutput( "Please, write the existing id of user" )]
+    if ( clickedBtn === CurrentButton.ShowAll )
+        content = [standardStrctr, table( clickedBtn, users, ifSbmtClicked )]
+    return DivWithClass( "containner",
+        DivWithClass( "row",
+            DivWithClass( "center-align",
```

Comment:
This is absolutely horrifying :-)

This works, but you don't need to have this huge if branching (also, why not use if-else?). What you could do is to render content as the exact same list of elements every time, e.g.
```js
content = [headerButtons(state), searchBar(state), resultsTable(state)];
```

As you can see, I give each component state object when rendering it. Now I can write those components in a way that **each element can decide, does IT need to be rendered?** E.g.
```ts
const resultsTable = (state: UserAppState) => {
   if (state.foundUsers.length < 1) {
      // No users found. We are not rendered!
      // If we return empty string, that pretty much means nothing will be added to HTML.
      return ""; 
   } else {
      // Actual code for rendering the table if there ARE some found users
   }
}
```

This way, you don't need to deconstruct each state scenario in one bulky function. You can instead create as many components as you like, render them, and give them current state so that they can make small decisions depending on their own needs.